packages/atclient/CMakeLists.txt

```diff
@@ -176,18 +191,25 @@ if(NOT ESP_PLATFORM)
   add_library(${PROJECT_NAME} STATIC ${atclient_srcs})
   add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
+  # Manually add header file
+  configure_file(${cjson_SOURCE_DIR}/cJSON.h ${CMAKE_BINARY_DIR}/include/cjson/cJSON.h)
+  
+  # Set include directories for atclient target
+  target_include_directories(${PROJECT_NAME} PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+    $<BUILD_INTERFACE:${ATCLIENT_INCLUDE_DIR}>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+  )
+
+  # atclient depends on cjson, so the latter must be built first
+  add_dependencies(atclient cjson)
```

Comment:
Doesn't have to be done in this PR, but should we also consider adding `atchops`, `uuid4` and `mbedtls` as dependencies to atclient?

Is this the magic line of code that fixed everything btw?

Reply:
For some reason, when I changed to `FetchContent_Declare` `cjson` was building later than the other packages, so I had to add that extra line. It didnt fix the issue, but was part of the fix

Reply:
I think we have a bunch of cleanup work to take care of in this file, but it can be dealt with later